OpenSim/MenuManager.swift

```diff
@@ -53,10 +54,39 @@ protocol MenuManagerDelegate {
     
     private func buildMenu() {
         let menu = NSMenu()
+
+        menu.addItem(NSMenuItem.separator())
+
+        let refreshMenuItem = menu.addItem(withTitle: UIConstants.strings.menuRefreshButton, action: #selector(self.refreshItemClicked(_:)), keyEquivalent: "r")
+        refreshMenuItem.target = self
+
+        let focusedModeMenuItem = menu.addItem(withTitle: UIConstants.strings.menuFocusedModeButton, action: #selector(self.toggleFocusedMode), keyEquivalent: "")
+        focusedModeMenuItem.target = self
+        focusedModeMenuItem.state = self.focusedMode ? .on : .off
+
+        let launchAtLoginMenuItem = menu.addItem(withTitle: UIConstants.strings.menuLaunchAtLoginButton, action: #selector(self.launchItemClicked(_:)), keyEquivalent: "")
+        launchAtLoginMenuItem.target = self
+        if existingItem(itemUrl: Bundle.main.bundleURL) != nil {
+            launchAtLoginMenuItem.state = .on
+        } else {
+            launchAtLoginMenuItem.state = .off
+        }
         
         DeviceManager.defaultManager.reload { (runtimes) in
-            runtimes.forEach { (runtime) in
-                let devices = runtime.devices.filter { $0.applications?.count ?? 0 > 0 }
+            
+            var sortedList = [Runtime]()
+            _ = Dictionary(grouping: runtimes, by: { (runtime: Runtime) in
+                return runtime.platform
+            }).values.map({ (runtimeList: [Runtime]) -> [Runtime] in
+                return runtimeList.sorted { $0.version ?? 0.0 > $1.version ?? 0.0 }
+            }).forEach({ (list) in
+                sortedList.append(contentsOf: list)
+            })
+            sortedList.forEach { (runtime) in
+                var devices = runtime.devices
+                if self.focusedMode {
+                    devices = devices.filter { $0.applications?.count ?? 0 > 0 }
```

Comment:
When I was using this, I missed the booted devices to also show up when the focused mode was active.
I had a booted simulator without any apps on it and I wanted to shut it down from the app but couldn't without turning focused mode off.
I personally think that we should also filter by booted devices, what do you think?